### ISSUE TITLE
Option to subset package overview plot

### ISSUE THREAD

**Issue by edward-burn** [darwin-eu-dev/PaRe#29]
To help aid refactoring, I think it would be nice if there was the ability to 
- limit package overview to a single function (with its ancestors and descendants)- this would help visualise how a specific function is being used in the package
- remove a sepecific function from the plot- this would help visualise the change to the package if I remove it

**Comment by mvankessel-EMC** [comment]
@edward-burn I've been messing around with this a little. I landed on the following implementation for now:

Given the following graph
```
nodes <- c(
  "A -> B",
  "A -> C",
  "A -> D",
  "A -> E",
  "D -> F",
  "B -> G",
  "C -> F",
  "F -> H"
)

DiagrammeR::grViz(
  paste0(
    "digraph {graph [layout = dot, rankdir = LR]",
    paste0(nodes, collapse = '\n'),
    "}"
  )
)
```
![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/f4c53f82-97ef-4105-89b5-7c23ffb39c4a)

Lets say we want to see the full path of node **C**.

We need to be able to go forward and backwards from that node.
```
forward <- function(c) {
  unlist(lapply(c, function(node) {
    start <- nodes[startsWith(nodes, node)]
    substr(start, nchar(start), nchar(start))
  }))
}

forward("C")
[1] "F"
```

```
backward <- function(c) {
  unlist(lapply(c, function(node) {
    start <- nodes[endsWith(nodes, c)]
    substr(start, 1, 1)
  }))
}

backward("C")
[1] "A"
```
Multiple nodes may be connected to a node, therefore we need to apply our implementation for each incoming or outgoing node.

From this point on we need to go forward, but recursively
```
recForward <- function(c) {
  f <- forward(c)
  if (length(f) > 0) {
    f <- unlist(lapply(f, recFor))
    return(paste(c, "->", f))
  } else {
    return(c)
  }
}

recForward("C")
[1] "C -> F -> H"
```

And recursively backwards
```
recBackward <- function(c) {
  f <- backward(c)
  if (length(f) > 0) {
    f <- unlist(lapply(f, recBack))
    return(paste(f, "->", c))
  } else {
    return(c)
  }
}

recBackward("C")
[1] "A -> C"
```

Then we use those in a function to plot our diagram:
```
funDiagram <- function(funName, nodes) {
  subFor <- paste0(recForward(funName), collapse = "\n")
  subBack <- paste0(recBackward(funName), collapse = "\n")
  
  DiagrammeR::grViz(
    paste0(
      "digraph {graph [layout = dot, rankdir = LR]",
      paste0(c(subFor, subBack), collapse = "\n"),
      "}"))
}

funDiagram("C", nodes)
```
![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/c4283624-5a77-4d8c-b8c7-32415be4f41f)

The only issue with this implementation is that it will plot duplicate paths, i.e. if we look for all paths of **A**.
```
funDiagram("A", nodes)
```

Because the following paths are produced:
```
# Forward from A
"A -> B -> G"
"A -> C -> F -> H"
"A -> D -> F -> H"
"A -> E"
# Backward from A
"A"
```

![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/1d477606-6c85-4966-9f91-2d4a73f44c38)

**Comment by mvankessel-EMC** [comment]
I've pushed an update to the dev branch for this. It now also handles looping paths, with a `maxIter` parameter supplied by the user.
![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/bd715a36-4ec5-4674-9658-fbb31e748b47)
_Fig 1. IncidencePrevalence_

![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/afc7b8b5-fee3-4cb2-bd69-c50b72b2c637)
_Fig 2. Sub path for `estimatePrevalence`_

Currently testing it CohortMethod where it behaves unexpectantly, but it seems to work for IncidnecePrevalence.

**Comment by edward-burn** [comment]
@mvankessel-EMC awesome, this looks great!

**Comment by mvankessel-EMC** [comment]
@edward-burn I just pushed an update to `dev`. I ended up using `igraph` to compute the incoming and outgoing paths from a specific node.

```
repo <- PaRe::Repository$new("../PaRe/")

PaRe::pkgDiagram(repo)
```
![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/ebe93959-9c61-4078-abdd-845896b120ec)

```
PaRe::getFunctionDiagram(repo, "funsUsedInLine")
```
![image](https://github.com/darwin-eu-dev/PaRe/assets/112690347/70ba4ec9-5058-427a-8d2d-8e4ccb70bd4f)


Could you try it out, report back, and let me know what you think?

**Comment by mvankessel-EMC** [comment]
Included in latest version 0.1.7 294bd2278ee1906d6a0768dcc545e855986ce79c
